enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,559 @@
+    for i, adm in enumerate(adm_loc):
+        if adm is not None:
+            location_description = adm['geometry']['label']
+    return f'Mousing over {location_description}'
+
+
+@APP.callback(
+    Output("loc_marker", "position"),
+    Output("lat_input", "value"),
+    Output("lng_input", "value"),
+    Input("submit_coords","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg["Y"][int(rr_mrg["Y"].size/2)].values
+        lng = rr_mrg["X"][int(rr_mrg["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+
+def crop_suitability(
+    rainfall,
+    min_wet_days,
+    wet_day_def,
+    tmax,
+    tmin,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    max_temp,
+    min_temp,
+    temp_range,
+    target_season,
+):
+    seasonal_precip = rainfall.sel(T=rainfall['T.season']==target_season)
+    seasonal_tmax = tmax.sel(T=tmax['T.season']==target_season)
+    seasonal_tmin = tmin.sel(T=tmin['T.season']==target_season)
+
+    seasonal_avg_tmax_suitability = 1*(
+        seasonal_tmax.groupby("T.year").mean() <= float(max_temp)
+    )
+    seasonal_avg_tmin_suitability = 1*(
+        seasonal_tmin.groupby("T.year").mean() >= float(min_temp)
+    )
+
+    seasonal_avg_temp_amplitude_suitability = 1*(
+        (seasonal_tmax - seasonal_tmin).groupby("T.year").mean() <= float(temp_range)
+    )
+    
+    seasonal_wet_days_suitability = 1*(
+        (seasonal_precip >= float(wet_day_def)).groupby("T.year").sum() >= float(min_wet_days)
+    )
+
+    seasonal_total_precip_suitability = 1*(
+        (seasonal_precip.groupby("T.year").sum() <= float(upper_wet_threshold)) &
+        (seasonal_precip.groupby("T.year").sum() >= float(lower_wet_threshold))
+    )
+    
+    crop_suitability = xr.Dataset(
+        data_vars = dict(
+        ),
+        coords = dict(
+            X = seasonal_avg_tmax_suitability["X"],
+            Y = seasonal_avg_tmax_suitability["Y"],
+            year = seasonal_avg_tmax_suitability["year"],
+        ), 
+    )
+    crop_suitability = crop_suitability.assign(
+        max_temp = seasonal_avg_tmax_suitability,
+        min_temp = seasonal_avg_tmin_suitability, 
+        temp_range = seasonal_avg_temp_amplitude_suitability,
+        precip_range = seasonal_total_precip_suitability,
+        wet_days = seasonal_wet_days_suitability,
+    )
+    crop_suitability['crop_suit'] = (
```

Comment:
I think here would be the appropriate place to convert the booleans to integers, rather than doing it earlier. `max_temp`, `min_temp`, etc. are naturally boolean values.